core/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -580,6 +582,28 @@ public synchronized ClientFileInfo getFileStatus(int fileId, boolean useCachedMe
     return getFileStatus(fileId, TachyonURI.EMPTY_URI, useCachedMetadata);
   }
 
+  private synchronized <K> ClientFileInfo getFileStatus(Map<K, ClientFileInfo> cache, K key,
+      int fileId, String path, boolean useCachedMetaData) throws IOException {
+    ClientFileInfo info = cache.get(key);
+    if (useCachedMetaData && info != null) {
+      return info;
+    }
+
+    info = mMasterClient.getFileStatus(fileId, path);
+    fileId = info.getId();
+    if (fileId == -1) {
```

Comment:
This block is to check whether the file is obsolete, if it is, we should clear cache related to it.

what is temp connection issue with master?